### ISSUE TITLE
[UX] Layout path selection: Make it more user friendly

### ISSUE THREAD

**Issue by docwilmot** [backdrop/backdrop-issues#2618]
From a Gitter conversation but similar sentiments at #2411 and #2306 

Sam Stamport @ColumbiaBlooms 07:51
Please explain the relationship between Path and Visibility Conditions on the layout configuration page in the UI.

...

docwilmot @docwilmot Apr 04 14:15
@Gormartsen idea:
On the Add layout page, offer the user two options (radio list): create a layout for existing entities (will need to work on the terminology) OR your own path
Selecting the first option would reveal a select list of entity types. Maybe clicking one could immediately reveal an option to pick a bundle type.
Selecting the second option reveals a textfield to type a path into
This wouldn't avoid confusion for complicated paths like other-entity/%/subpath/% but would at least simplify the majority of common use cases.
Thoughts?

docwilmot @docwilmot 13:16
Add layout page
![layoutadd](https://cloud.githubusercontent.com/assets/1042283/24729028/06c03e76-1a22-11e7-98e3-b3bf3c7ec051.PNG)
Entering custom path
![layoutpath](https://cloud.githubusercontent.com/assets/1042283/24729039/0bd42c60-1a22-11e7-8e07-99933904f3c7.PNG)
Selecting an entity to override its paths.
![layoutentity](https://cloud.githubusercontent.com/assets/1042283/24729043/0d838524-1a22-11e7-884a-ad6c85ae65a1.PNG)

Gregory Netsas @klonos 17:33
@docwilmot I personally like the idea of making the process of creating layouts be wizard-based. I wonder if adding an additional step (and thus increasing the time/clicks required to create a layout) will disappoint advanced users. On the other hand, we could add this as a "layout path wizard" next to the path field. Advanced users would still be able to enter the path as they used to, but there will be a "Help me with this" button that pops-up a dialog next to it for users that do not understand how this works.

**Comment by docwilmot** [comment]
In summary: the add layout page could use some UX attention, probably a wizard, to make it less terrifying.

**Comment by klonos** [comment]
@docwilmot thanx for opening this issue here. I have spent the past few hours re-reading the other related issues and pondering.

I am curious as to why in your mockups you chose the word "save template" when creating layouts for entities.

Also why "Choose your **target**"?

If this is to become a wizard, I'd expect it to offer all relevant choices, and if meant to be less intimidating, avoid using the words "node" and "entity". So I was thinking something like a radio set with label "Create layout for:" and these options:

- Content
- Comments
- Users
- Taxonomy terms
- Manually enter a path (advanced)

**Comment by docwilmot** [comment]
@klonos this was a quick mockup with random words for demo only.

**Comment by klonos** [comment]
Fair enough, then it looks good for a starting point. What do you think about offering all available options as a list, with the (advanced) manual path entry being last in the list?

**Comment by docwilmot** [comment]
> What do you think about offering all available options as a list, with the (advanced) manual path entry being last in the list?

Thats a better idea, yes.

**Comment by jenlampton** [comment]
We're 2 weeks from code freeze and without code here to review or revise its not likely this will make it in. I'm going to bump the milestone to 1.9.

**Comment by klonos** [comment]
We brought this up during the dev meeting today. Inspired by the Views "Add view" wizard, here's how this could look:

![image](https://user-images.githubusercontent.com/2423362/90830904-3fc2ba00-e34b-11ea-9875-ddf788d05268.png)

(pinging @stpaultim for thoughts ^^)

**Comment by docwilmot** [comment]
This is on creation, what would it look like on edit?

**Comment by jenlampton** [comment]
I love it :) even better without the words "node" and "entity" (I was
wondering how we were going to do that!)

**Comment by stpaultim** [comment]
@klonos - This is a great start and helps me better understand what you were getting at. 

I think that maybe we could include visibility rules on this initial page as well. I think they are pretty basic and often the reason I am creating a new layout.

**Comment by klonos** [comment]
> This is on creation, what would it look like on edit?

I'll work on a mockup for that too 👍 

Glad you guys like it 🙂

**Comment by klonos** [comment]
> I think that maybe we could include visibility rules on this initial page as well. I think they are pretty basic and often the reason I am creating a new layout.

Yes. See how in my mockup there's a second select after the first one, with label "of type"? ...there's your basic visibility condition set right there 😉 :

![Screen Shot 2020-08-21 at 3 55 50 am](https://user-images.githubusercontent.com/2423362/90840365-7ce67680-e362-11ea-8f66-de0362190719.png)

...the "Add view" wizard works similarly, by adding basic filters to the view:

![image](https://user-images.githubusercontent.com/2423362/90840458-b323f600-e362-11ea-8855-a7df20688e37.png)
